### PR TITLE
fix: use axios instance instead of global instance

### PR DESCRIPTION
**Pull request by menthol**
I try to use axios instance instead of global instance to not overwrite axios global config.

### CODE REVIEW

**Comment by menthol** [comment]
Hello,

This is a big change. Using a dedicated Axios instance allows you to gain interesting functionality, having multiple instances of Kitsu that have different URIs.

This change also avoids touching the default instance of Axios and avoids breaking applications that already have an API and are gradually becoming JsonApi.

**This change may break existing applications if the application uses the fact that Kitsu modifies the base URL of the default Axios instance to use their own API.**

File: src/get.spec.js [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import axios from 'axios'
```
Comment (wopian) [inline src/get.spec.js]: Mocking `api.axios` is not needed as `axios-mock-adapter` directly hooks onto  the `axios()` called by the instance.

Comment (menthol) [reply src/get.spec.js]: I do not understand why mocking `api.axios` it is not necessary, Kitsu now has a different instance of Axios for each of his instances. If they do not mock it, I do not see how I could run the tests. Maybe I lack knowledge on the subject.

Comment (wopian) [reply src/get.spec.js]: The mocker only creates fake responses for specific routes (i.e `GET /users`). Axios itself (in this case the instance) still operates the same, so the headers are applied to the instance and sent correctly.

Thus, creating a single mock is functionally equivalent to creating a mock for each test over and over.

**Comment by menthol** [comment]
Seem to work, do you want me to make a single commit pull request ?

**Comment by wopian** [comment]
Released in `4.3.0`

**Pull request merged**